newa/__init__.py

```diff
@@ -484,18 +484,28 @@ class Recipe(Cloneable, Serializable):
 RecipeEnvironment = dict[str, str]
 
 
+class RawRecipeReportPortalConfigDimension(TypedDict, total=False):
```

Comment:
Ugh, that is quite a long name :). Why it is called raw? Can't we have it defined in a recipe class so that we can simply call it is ReportPortalConfg?

Reply:
It is being used in Recipe and  RawRecipeConfigDimension classes. Maybe we could define it under the Recipe class, TBH I don't know what's better and will follow other's recommendations. 
@happz Opinions?

Reply:
Raw prefix originates from Milos's code so I will leave the answer on him.

Reply:
"Raw" as in "user or service input". Often this input is then processed and transformed into better-fitting data structures.

There's no perfect solution... You want the various mappings and lists entering from YAML files to be described by typed dicts and lists and properly annotated, so the code would know that this particular mapping has these keys, and avoid typos or treating a list as a dictionary. But there's a price, as mappings and lists get nested deeper in each other, components like this one tend to get longer names, expressing the hierarchy, "a dimension of report portal configuration in a recipe"...

What I would do: I'd add a comment, or adopt a naming pattern, like `Raw_Recipe_ReportPortal_Config_Dimension`, to separate levels, expressing the hierarchy, and check it has the right fields and it's at the right place. You may also decide to drop all these `TypedDict`, but then you risk letting the structure to grow out of control.

I don't like relying on nested dictionaries and lists exactly because of this: it's so easy to start with, one can prototype quickly and ship something working. It's so tempting, it's easy to have seven levels of dictionaries in a YAML file. But then RFEs start coming, and the dictionaries start getting new keys and nest even more, and suddenly, a list appears where I expected a string and everything blows out somewhere in production. New contributors start adding new things, and this structure grows, unguarded, and wild... There's no validation. Using `TypedDict` to define partial types and then compose bigger structures from them is the only defense I know, plus a schema-based validation. Or defining proper data classes, and use `attr` to load data from YAML and materialize it as a tree of genuine Python objects rather than raw dictionaries.

Reply:
Hi @The-Mule ,  given the options above, what would you prefer?

Reply:
Thanks @happz for clarification. It makes sense, alright I suppose we can leave it as it is, no need to rename it.